### PR TITLE
fix: handle forbidden squash/rebase/merge commits

### PR DESCRIPTION
**Pull request by sileht**
Fixes MRGFY-266

### CODE REVIEW

File: mergify_engine/actions/merge_base.py [hunk 1]
```diff
@@ -49,6 +49,7 @@
 MARKDOWN_TITLE_RE = re.compile(r"^#+ ", re.I)
 MARKDOWN_COMMIT_MESSAGE_RE = re.compile(r"^#+ Commit Message ?:?\s*$", re.I)
 REQUIRED_STATUS_RE = re.compile(r'Required status check "([^"]*)" is expected.')
+FORBIDDEN_MERGE_COMMITS_MSG = "Merge commits are not allowed on this repository."
```
Comment (jd) [inline mergify_engine/actions/merge_base.py:52]: I'm pretty sure we should also handle `squash` or `rebase` not being allowed.

File: mergify_engine/actions/merge_base.py [hunk 2]
```diff
@@ -509,6 +510,20 @@ async def _handle_merge_error(
...
+            elif FORBIDDEN_MERGE_COMMITS_MSG in e.message:
+                ctxt.log.info(
+                    "Repository configuration doesn't allow merge commit",
+                    status_code=e.status_code,
+                    error_message=e.message,
+                )
+                return check_api.Result(
+                    check_api.Conclusion.CANCELLED,
+                    e.message,
+                    "The repository configuration doesn't allow merge commit. "
```
Comment (jd) [inline mergify_engine/actions/merge_base.py:522]:
```suggestion
                    "The repository configuration doesn't allow merge commits. "
```

File: mergify_engine/actions/merge_base.py [hunk 3]
```diff
@@ -509,6 +510,20 @@ async def _handle_merge_error(
...
+                    "The merge `method` configured in Mergify configuration must be "
+                    "allowed in the repository configuration setting",
```
Comment (jd) [inline mergify_engine/actions/merge_base.py:524]:
```suggestion
                    "allowed in the repository configuration settings.",
```

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/actions/merge_base.py (hunk 1 above)
Comment (sileht) [reply mergify_engine/actions/merge_base.py:52]: I will add them.

**Review by jd — APPROVED** [review]

**Comment by sileht** [comment]
@Mergifyio update

**Pull request merged**